Mathlib/NumberTheory/LSeries/HurwitzZetaEven.lean

```diff
@@ -0,0 +1,386 @@
+  rw [cosKernel_def a t]
+  convert hasSum_jacobiTheta₂_term a (by rwa [I_mul_im, ofReal_re] : 0 < im (I * t)) using 2 with n
+  rw [jacobiTheta₂_term, ofReal_exp, ← Complex.exp_add]
+  push_cast
+  ring_nf
+  simp only [I_sq, mul_neg, neg_mul, mul_one, sub_eq_add_neg]
+
+/-- Modified version of `hasSum_int_evenKernel` omitting the constant term at `∞`. -/
+lemma hasSum_int_evenKernel₀ (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℤ ↦ if n + a = 0 then 0 else rexp (-π * (n + a) ^ 2 * t))
+    (evenKernel a t - if (a : UnitAddCircle) = 0 then 1 else 0) := by
+  simp_rw [AddCircle.coe_eq_zero_iff, zsmul_one]
+  split_ifs with h
+  · obtain ⟨k, rfl⟩ := h
+    simp_rw [← Int.cast_add, Int.cast_eq_zero, add_eq_zero_iff_eq_neg]
+    simpa using hasSum_ite_sub_hasSum (hasSum_int_evenKernel (k : ℝ) ht) (-k)
+  · suffices ∀ (n : ℤ), n + a ≠ 0 by simpa [this] using hasSum_int_evenKernel a ht
+    contrapose! h
+    let ⟨n, hn⟩ := h
+    exact ⟨-n, by rwa [Int.cast_neg, neg_eq_iff_add_eq_zero]⟩
+
+lemma hasSum_int_cosKernel₀ (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℤ ↦ if n = 0 then 0 else cexp (2 * π * I * a * n) * rexp (-π * n ^ 2 * t))
+    (↑(cosKernel a t) - 1) := by simpa using hasSum_ite_sub_hasSum (hasSum_int_cosKernel a ht) 0
+
+lemma hasSum_nat_cosKernel₀ (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℕ ↦ 2 * Real.cos (2 * π * a * (n + 1)) * rexp (-π * (n + 1) ^ 2 * t))
+    (cosKernel a t - 1) := by
+  rw [← hasSum_ofReal, ofReal_sub, ofReal_one]
+  have := (hasSum_int_cosKernel a ht).sum_nat_of_sum_int
+  rw [← hasSum_nat_add_iff' 1] at this
+  simp_rw [Finset.sum_range_one, Nat.cast_zero, neg_zero, Int.cast_zero, zero_pow two_ne_zero,
+    mul_zero, zero_mul, Complex.exp_zero, Real.exp_zero, ofReal_one, mul_one, Int.cast_neg,
+    Int.cast_natCast, neg_sq, ← add_mul, add_sub_assoc, ← sub_sub, sub_self, zero_sub,
+    ← sub_eq_add_neg, mul_neg] at this
+  convert this with n
+  push_cast
+  rw [Complex.cos, mul_div_cancel₀ _ two_ne_zero]
+  ring_nf
+
+/-!
+## Asymptotics of the kernels as `t → ∞`
+-/
+
+/-- The function `evenKernel a - L` has exponential decay at `+∞`, where `L = 1` if
+`a = 0` and `L = 0` otherwise. -/
+lemma isBigO_atTop_evenKernel_sub (a : UnitAddCircle) : ∃ r : ℝ, 0 < r ∧
+    (evenKernel a · - (if a = 0 then 1 else 0)) =O[atTop] (rexp <| -r * ·) := by
+  obtain ⟨b, _, rfl⟩ := a.eq_coe_Ico
+  obtain ⟨p, hp, hp'⟩ := HurwitzKernelBounds.isBigO_atTop_F_int_zero_sub b
+  refine ⟨p, hp, (EventuallyEq.isBigO ?_).trans hp'⟩
+  filter_upwards [eventually_gt_atTop 0] with t ht
+  simp only [← (hasSum_int_evenKernel b ht).tsum_eq, HurwitzKernelBounds.F_int,
+    HurwitzKernelBounds.f_int, pow_zero, one_mul, Function.Periodic.lift_coe]
+
+/-- The function `cosKernel a - 1` has exponential decay at `+∞`, for any `a`. -/
+lemma isBigO_atTop_cosKernel_sub (a : UnitAddCircle) :
+    ∃ p, 0 < p ∧ IsBigO atTop (cosKernel a · - 1) (fun x ↦ Real.exp (-p * x)) := by
```

Comment:
Is there a reason to call the positive real whose existence is asserted ` r`  above and ` p`  here?